### ISSUE TITLE
Loss of the geometry of a vector layer

### ISSUE THREAD

**Issue by mwohlNCA** [qgis/QGIS#41964]
After modifying and saving a layer with edit mode, the geometry of the layer disappears. Only the attribute table remains. 
 
1. Load a vector layer in QGIS
2. Start the edit mode on this layer
3. Digitize new vector entities
4. Save the modifications
5. Quit the edit mode
6. The following error message appears:
"Cannot zoom in on selected entities: No extent could be determined"

QGIS version
3.10.14-A Coruña
Code review
8374282d2a
Compiled with Qt
5.11.2
Using Qt
5.11.2
Compiled with GDAL / OGR
3.1.4
Used with GDAL / OGR
3.1.4
Compiled with GEOS
3.8.1-CAPI-1.13.3
Used with GEOS
3.8.1-CAPI-1.13.3
Compiled with SQLite
3.29.0
Works with SQLite
3.29.0
PostgreSQL client version
11.5
SpatiaLite version
4.3.0
QWT version
6.1.3
QScintilla2 version
2.10.8
Compiled with PROJ
6.3.2
Works with PROJ
Rel. 6.3.2, May 1st, 2020
OS version
Windows 10 (10.0)
Active Python extensions
cartoini;
pluginbuilder3;
plugin_reloader;
QuickOSM;
quick_map_services;
db_manager;
MetaSearch;
processing

This issue occurs when the layer is saved on the corporate server. If the layer is saved locally on a computer, there is no such problem. Other companies have already encountered this same problem but without having found the solution.
![image_2021_02_22T16_37_16_197Z](https://user-images.githubusercontent.com/79859135/109628412-01e8ee80-7b43-11eb-8813-77e30558982d.png)

**Comment by rduivenvoorde** [comment]
@mwohlNCA can you be more specific:

What kind of 'vector layer'?  QGIS has A LOT of different types of vector layers...

What do you mean with "the layer is saved on the corporate server"? 
Are you talking about a network drive (what kind: windows/smb? nfs? sharepoint? google-like drive?

**Comment by gioman** [comment]
>  Are you talking about a network drive (what kind: windows/smb? nfs? sharepoint? google-like drive?

@mwohlNCA or maybe is a database (postgis? oracle?)?

>  Other companies

@mwohlNCA what other companies?

**Comment by mwohlNCA** [comment]
Hello. We work with layers in shapefile format
In our company we use a server which gathers all our GIS data. We encounter the problem when shapefile layers are saved on this server. I am attaching a capture of the organization of our server. Thank you for your help 🙂

Mathys WOHL
NCA Environnement
________________________________
De : Richard Duivenvoorde <notifications@github.com>
Envoyé : mardi 2 mars 2021 12:20
À : qgis/QGIS <QGIS@noreply.github.com>
Cc : Mathys Wohl SIG <m.wohl@nca.fr>; Mention <mention@noreply.github.com>
Objet : Re: [qgis/QGIS] Loss of the geometry of a vector layer (#41964)


@mwohlNCA<https://github.com/mwohlNCA> can you be more specific:

What kind of 'vector layer'? QGIS has A LOT of different types of vector layers...

What do you mean with "the layer is saved on the corporate server"?
Are you talking about a network drive (what kind: windows/smb? nfs? sharepoint? google-like drive?

**Comment by gioman** [comment]
> We encounter the problem when shapefile layers are saved on this server.

@mwohlNCA do you mean a shared network folder? if yes what type pf share is (samba? else?)?

Do you have the same issue when the shapefiles are stored on a local drive?



> I am attaching a capture of the organization of our server.

You didn't attached anything.

**Comment by mwohlNCA** [comment]
Yes i talk about  shared network folder but i can't give more informations about the server 😕
No we only have the issue when the layer was saved on the server. In local drive we don't have the problem.
I try agian to join the screenshot of the organisation of our server.

________________________________
De : Giovanni Manghi <notifications@github.com>
Envoyé : mardi 2 mars 2021 14:54
À : qgis/QGIS <QGIS@noreply.github.com>
Cc : Mathys Wohl SIG <m.wohl@nca.fr>; Mention <mention@noreply.github.com>
Objet : Re: [qgis/QGIS] Loss of the geometry of a vector layer (#41964)


We encounter the problem when shapefile layers are saved on this server.

@mwohlNCA<https://github.com/mwohlNCA> do you mean a shared network folder? if yes what type pf share is (samba? else?)?

Do you have the same issue when the shapefiles are stored on a local drive?

I am attaching a capture of the organization of our server.

You didn't attached anything.

**Comment by gioman** [comment]
>  i can't give more informations about the server

@mwohlNCA well. knowing the type of share could be an important piece of information to confirm/troubleshoot the problem. Ask your IT guys.

**Comment by roya0045** [comment]
I get this sometimes but reloading the project works fine. The tables empties on connection issues and repopulate on project reopening.

**Comment by mwohlNCA** [comment]
Hello. I have more informations about our server. It’s Windows 2012 R2 Standard He has 3 roles : Files, DFS and printing.
Thank you for your help.

Mathys WOHL

De : Giovanni Manghi ***@***.***>
Envoyé : mardi 2 mars 2021 15:04
À : qgis/QGIS ***@***.***>
Cc : Mathys Wohl SIG ***@***.***>; Mention ***@***.***>
Objet : Re: [qgis/QGIS] Loss of the geometry of a vector layer (#41964)


i can't give more informations about the server

@mwohlNCA<https://github.com/mwohlNCA> well. knowing the type of share could be an important piece of information to confirm/troubleshoot the problem. Ask your IT guys.

**Comment by ninsbl** [comment]
A colleague of mine is experiencing this issue after digitizing on a OneDrive directory, that is synced periodically and where sometimes syncing in the background may fail if authentication tokens are no longer valid.
But I did not experience this on NFS or CIFS shares...
@mwohlNCA do you know (or could you find out) if the directory on your server is synced to OneDrive? There are a couple of posts on both stackoverflow and microsoft Q&A that report syncing issues with OneDrive leading to corrupt files.

You may try the answer here: https://gis.stackexchange.com/questions/373984/backed-up-shapefile-is-no-longer-a-valid-or-recognized-data-source to recover your data. For my colleague I reverted to earlier versions on OneDrive. But in general I think staying away from OneDrive for digitizing is something I would carefully consider...

**Comment by rduivenvoorde** [comment]
One thought here: a 'shapefile' off course consists of several files (shp, dbf etc etc). I'm not sure if the synching machinery is aware of that, OR that it keeps the individual files 'in synch' but thereby messing up the 'complete shp'.

IF this is a case (I'm just guessing), a work around would be to not use shp files, but geopackages?
OR (even better) start using a proper spatial database like Postgis... You can always export your data to shp files if that is what your org is used to?

**Comment by gioman** [comment]
I changed the title and replaced "network" with "cloud" as this problem do not affects samba/nfs/etc. but probably only dropbox/onedrive/etc. (anyway I also think that no one should edit a shapfile being synchronized with one of those services).

**Comment by thbaumann** [comment]
> I changed the title and replaced "network" with "cloud" as this problem do not affects samba/nfs/etc. but probably only dropbox/onedrive/etc. (anyway I also think that no one should edit a shapfile being synchronized with one of those services).

In this comment the OP mentioned that the shapefile is stored an a windows server with file role, so pretty sure samba:
https://github.com/qgis/QGIS/issues/41964#issuecomment-802020988

At my workplace we also have a lot of problems with QGIS 3.16.5 and shapefiles stored on samba shares at the moment. 
(not neccessary to mention that still using shapefiles in the year 2021 is "suboptimal"... that's not my decision for the affected users.switching to postgis is already planned)

The users only access their own shapefiles but still there are problems with corrupted shapefiles (repack errors), NULL-geometries as result of using tools like split feature / line offset (probably fixed soon in https://github.com/libgeos/geos/issues/477 )  and confused snapping-tool (snaps to wrong segments) or vertex tool (not showing certain vertices of line features).

So I guess it's more a problem of the antiquated shapefile format and it's restrictions which occurs more when they are accessed from network drives.

**Comment by thbaumann** [comment]
For my testcase the bug is not reproducible anymore if I set following registry keys to 0 :
DirectoryCacheLifetime, FileInfoCacheLifetime and FileNotFoundCacheLifetime ( HKEY_LOCAL_MACHINE\System\CurrentControlSet\Services\Lanmanworkstation\Parameters ) .
ESRI also had problems with the SMB redirector cache in the past and the Win10LTSSC2019 seems to have a similar problem which can be fixed by setting these registry keys.
https://support.esri.com/en/technical-article/000011778
